src/fabric/src/fabric_drop_seq.erl

```diff
@@ -0,0 +1,1016 @@
+                                        false
+                                end
+                        end
+                    end,
+                    ShardsInRange
+                )
+            ),
+            maps:merge_with(fun merge_peers/3, AsMap, Acc)
+        end,
+        #{},
+        PeerCheckpoints
+    ).
+
+create_peer_checkpoint_doc_if_missing(
+    <<"shards/", _/binary>> = DbName, Subtype, Source, PeerId, UpdateSeq
+) when
+    is_binary(DbName), is_binary(PeerId), is_integer(UpdateSeq)
+->
+    create_peer_checkpoint_doc_if_missing(
+        DbName, Subtype, Source, PeerId, pack_seq(DbName, UpdateSeq)
+    );
+create_peer_checkpoint_doc_if_missing(
+    DbName, Subtype, Source, PeerId, UpdateSeq
+) when
+    is_binary(DbName),
+    is_binary(Subtype),
+    is_binary(Source),
+    is_binary(PeerId),
+    is_integer(UpdateSeq)
+->
+    ok;
+create_peer_checkpoint_doc_if_missing(
+    <<"shards/", _/binary>> = DbName, Subtype, Source, PeerId, UpdateSeq
+) when
+    is_binary(DbName),
+    is_binary(Subtype),
+    is_binary(Source),
+    is_binary(PeerId),
+    is_binary(UpdateSeq)
+->
+    {_, Ref} = spawn_monitor(fun() ->
+        case
+            fabric:open_doc(mem3:dbname(DbName), peer_checkpoint_id(Subtype, PeerId), [?ADMIN_CTX])
+        of
+            {ok, _} ->
+                ok;
+            {not_found, _} ->
+                update_peer_checkpoint_doc(DbName, Subtype, Source, PeerId, UpdateSeq);
+            {error, Reason} ->
+                throw({checkpoint_commit_failure, Reason})
+        end
+    end),
+    receive
+        {'DOWN', Ref, _, _, ok} ->
+            ok;
+        {'DOWN', Ref, _, _, Else} ->
+            Else
+    end.
+
+update_peer_checkpoint_doc(
+    <<"shards/", _/binary>> = DbName, Subtype, Source, PeerId, UpdateSeq
+) when
+    is_binary(DbName),
+    is_binary(Subtype),
+    is_binary(Source),
+    is_binary(PeerId),
+    is_integer(UpdateSeq)
+->
+    update_peer_checkpoint_doc(DbName, Subtype, Source, PeerId, pack_seq(DbName, UpdateSeq));
+update_peer_checkpoint_doc(
+    DbName, Subtype, Source, PeerId, UpdateSeq
+) when
+    is_binary(DbName),
+    is_binary(Subtype),
+    is_binary(Source),
+    is_binary(PeerId),
+    is_integer(UpdateSeq)
+->
+    ok;
+update_peer_checkpoint_doc(
+    <<"shards/", _/binary>> = DbName, Subtype, Source, PeerId, UpdateSeq
+) when
+    is_binary(DbName),
+    is_binary(Subtype),
+    is_binary(Source),
+    is_binary(PeerId),
+    is_binary(UpdateSeq)
+->
+    {_, OpenRef} = spawn_monitor(fun() ->
+        case
+            fabric:open_doc(mem3:dbname(DbName), peer_checkpoint_id(Subtype, PeerId), [?ADMIN_CTX])
+        of
+            {ok, ExistingDoc} ->
+                exit(ExistingDoc#doc.revs);
+            {not_found, _Reason} ->
+                exit({0, []});
+            {error, Reason} ->
+                throw({checkpoint_fetch_failure, Reason})
+        end
+    end),
+    receive
+        {'DOWN', OpenRef, _, _, Revs} ->
```

Comment:
The shape of this message is the same as the one on line 530. You probably want to check the shape of Revs. Otherwise the `{checkpoint_fetch_failure, Reason}` would end up in the term.

```erlang
17> {_, OpenRef} = spawn_monitor(fun() -> throw(fff) end).
{<0.112.0>,#Ref<0.2499101716.2367684609.212273>}
=ERROR REPORT==== 18-Jun-2025::12:09:20.981098 ===
Error in process <0.112.0> with exit value:
{{nocatch,fff},[{shell,apply_fun,3,[{file,"shell.erl"},{line,914}]}]}

18> flush().
Shell got {'DOWN',#Ref<0.2499101716.2367684609.212273>,process,<0.112.0>,
                  {{nocatch,fff},
                   [{shell,apply_fun,3,[{file,"shell.erl"},{line,914}]}]}}
```